File: apps/api/app/schemas/schedule.py
from datetime import date, time

from pydantic import BaseModel, Field

from app.common.enums import ScheduleStatus
from app.schemas.common import AuditMetadataSchema


class ScheduleCreateSchema(BaseModel):
    experience_id: str
    date: date
    start_time: time
    is_active: bool = True
    capacity_total: int = Field(gt=0)
    reserved_slots: int = Field(default=0, ge=0)
    held_slots: int = Field(default=0, ge=0)
    internal_slots: int = Field(default=0, ge=0)
    blocked_slots: int = Field(default=0, ge=0)
    custom_request_only: bool = False
    notes: str | None = None


class ScheduleUpdateSchema(BaseModel):
    is_active: bool | None = None
    capacity_total: int | None = Field(default=None, gt=0)
    reserved_slots: int | None = Field(default=None, ge=0)
    held_slots: int | None = Field(default=None, ge=0)
    internal_slots: int | None = Field(default=None, ge=0)
    blocked_slots: int | None = Field(default=None, ge=0)
    custom_request_only: bool | None = None
    notes: str | None = None
    status: ScheduleStatus | None = None


class ScheduleResponseSchema(AuditMetadataSchema):
    id: str
    experience_id: str
    date: date
    start_time: time
    is_active: bool
    capacity_total: int
    reserved_slots: int
    held_slots: int
    internal_slots: int
    blocked_slots: int
    available_slots: int
    status: ScheduleStatus
    custom_request_only: bool
    notes: str | None

File: apps/mobile/lib/features/catalogs/schedules/domain/schedule.dart
import 'package:mobile/features/catalogs/data/catalog_sync_status.dart';
import 'schedule_status.dart';

class CatalogSchedule {
  const CatalogSchedule({
    required this.id,
    required this.experienceId,
    required this.date,
    required this.startTime,
    required this.isActive,
    required this.capacityTotal,
    required this.reservedSlots,
    required this.internalSlots,
    required this.blockedSlots,
    required this.availableSlots,
    required this.status,
    required this.customRequestOnly,
    required this.syncStatus,
    this.notes,
    this.remoteId,
    this.versionRemote,
    this.syncError,
    this.updatedAtRemote,
  });

  final String id;
  final String? remoteId;
  final String experienceId;
  final String date;
  final String startTime;
  final bool isActive;
  final int capacityTotal;
  final int reservedSlots;
  final int internalSlots;
  final int blockedSlots;
  final int availableSlots;
  final CatalogScheduleStatus status;
  final bool customRequestOnly;
  final String? notes;
  final CatalogSyncStatus syncStatus;
  final int? versionRemote;
  final String? syncError;
  final DateTime? updatedAtRemote;
}
